packages/sit-onyx/src/utils/numbers.spec.ts

```diff
@@ -55,3 +61,37 @@ describe("roundToPrecision", () => {
     expect(roundToPrecision(0, 2)).toBe("0.00");
   });
 });
+
+describe("convertBinaryPrefixToBytes", () => {
+  it.each<{ value: BinaryPrefixedSize; expected: number }>([
+    { value: "4.25Ki", expected: 4.25 * 1024 ** 1 },
+    { value: "4.25Mi", expected: 4.25 * 1024 ** 2 },
+    { value: "4.25Gi", expected: 4.25 * 1024 ** 3 },
+    { value: "4.25Ti", expected: 4.25 * 1024 ** 4 },
+    { value: "4.25Pi", expected: 4.25 * 1024 ** 5 },
+    { value: "4.25Ei", expected: 4.25 * 1024 ** 6 },
+    { value: "4.25Zi", expected: 4.25 * 1024 ** 7 },
+    { value: "4.25Yi", expected: 4.25 * 1024 ** 8 },
+    { value: "4.25Ri", expected: 4.25 * 1024 ** 9 },
+    { value: "4.25Qi", expected: 4.25 * 1024 ** 10 },
+  ])("should convert binary prefix $value to $expected bytes", ({ value, expected }) => {
+    const bytes = convertBinaryPrefixToBytes(value);
+    expect(bytes).toBe(expected);
+  });
+});
+
+describe("formatBytesToString", () => {
+  it.each<{ value: number; expected: string }>([
+    { value: 0, expected: "0B" },
+    { value: 1, expected: "1B" },
+    { value: 1024, expected: "1kB" },
+    { value: 1024 ** 2, expected: "1MB" },
+    { value: 1024 ** 3, expected: "1GB" },
+    { value: 1024 ** 4, expected: "1TB" },
+    { value: 1024 ** 5, expected: "1PB" },
+    { value: 1024 ** 6, expected: "1,024PB" },
```

Comment:
Correctly it would be the following:
```suggestion
    // Decimal-base bytes
    { value: 0, expected: "0B" },
    { value: 1, expected: "1B" },
    { value: 1000, expected: "1kB" },
    { value: 1000 ** 2, expected: "1MB" },
    { value: 1000 ** 3, expected: "1GB" },
    { value: 1000 ** 4, expected: "1TB" },
    { value: 1000 ** 5, expected: "1PB" },
    { value: 1000 ** 6, expected: "1,000PB" },

    // Binary Prefixes
    { value: 0, expected: "0B" },
    { value: 1, expected: "1B" },
    { value: 1024, expected: "1kiB" },
    { value: 1024 ** 2, expected: "1MiB" },
    { value: 1024 ** 3, expected: "1GiB" },
    { value: 1024 ** 4, expected: "1TiB" },
    { value: 1024 ** 5, expected: "1PiB" },
    { value: 1024 ** 6, expected: "1,024PiB" },
```


But if Intl doesn't support binary prefixes, we can also stay with decimal based units.

Reply:
Unfortunately not. [Here](https://tc39.es/proposal-unified-intl-numberformat/section6/locales-currencies-tz_proposed_out.html#sec-issanctionedsimpleunitidentifier) is a list of supported units.

Also since this formatting is used to display the size to the (mostly non-technical) user, I think "1MiB" is more confusing to the user than "1MB" so I'd like to keep this formatting